libs/komenci-logger/src/komenci-logger.service.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable, LoggerService } from '@nestjs/common'
+import { Logger } from "nestjs-pino"
+
+import {
+    AttestationsRequested,
+    KomenciEvent,
+    MetaTransactionSubmitted,
+    PepperRequested,
+    RuleVerified,
+    TxEvent,
+    WalletDeployed
+} from '@app/komenci-logger/events'
+
+export interface KomenciLogger extends LoggerService {
+    logSuccessfulSessionStart: (sessionStartFailEvent: KomenciEvent) => void
+    logFailedSessionStart: (sessionStartFailEvent: KomenciEvent) => void
+    logRuleChecked: (ruleVerifiedEvent: RuleVerified) => void
+    logWalletDeployed: (walletDeployEvent: WalletDeployed) => void
+    logBlindedPepperRequest: (pepperRequestEvent: PepperRequested) => void
+    logCompletedTransaction: (txEvent: TxEvent) => void
+    logSubmittedMetaTransaction: (submittedMTx: MetaTransactionSubmitted) => void
+    logSubsidizedAttestations: (attestationsRequested: AttestationsRequested) => void
+}
+
+@Injectable()
+export class KomenciLoggerService implements KomenciLogger {
```

Comment:
We should have this service be request scoped and inject the request in the constructor. That way we can extract the session related info from the request and not have that be passed in for each event.
Check [this page in the docs](https://docs.nestjs.com/fundamentals/injection-scopes) for info on request scoped services and [this specific section](https://docs.nestjs.com/fundamentals/injection-scopes#request-provider) on using the request provider in order to get the request on service init.